Update Fetch Migration documentation (#393)
opensearch-project/opensearch-migrations: FetchMigration/README.md

# Fetch  Migration and Backfill

The **Fetch Migration** tool simplifies the process of moving indexes and the data contained within them from a
source cluster, such as Elasticsearch or OpenSearch, to a target OpenSearch cluster. It automates the process of 

> comparing indexes between the two clusters by creating index metadata, like settings and mappings that does not already exist on the target cluster using [Data Prepper](https://github.com/opensearch-project/data-prepper).

Reviewer: There are some technical inaccuracies here:
1. Data Prepper is only used for data migration, and not for index comparison and creation as this line implies
2. Indexes are compared _before_ they are created, not _by_ creating index metadata.

> The Fetch Migration tool consists of a Python script. However, a Docker image can be built using the included [Dockerfile](./Dockerfile) included in this repository.

Author: ```suggestion
The Fetch Migration tool consists of a Python script which requires Python 3 or greater. However, a Docker image can be built using the included [Dockerfile](./Dockerfile) included in this repository.
```

> The Fetch Migration tool consists of a Python script which requires Python 3 or greater. However, a Docker image can be built using the included [Dockerfile](./Dockerfile) included in this repository.

Reviewer: Nit - "However" seems unnecessary here? It can be removed.

## Components

The tool consists of three components:

* The **Metadata migration** module handles metadata comparison between the source and target clusters. This can output a human-readable report as well as a Data Prepper `pipeline.yaml` file.
* The **Migration monitor** module monitors the progress of the migration and shuts down the Data Prepper pipeline 
once the target document count has been reached
* The **Orchestrator** module that sequences these steps as a workflow and manages the kick-off of the Data Prepper 
process between them. The orchestrator module helps route the correct components when using Docker, though each component can be executed seperately using Python.

Help text for each module can be printed using the `-h / --help` flag when exectuing the Python script.

## Current Limitations

The Fetch Migration currently has the following limitations:

* Fetch Migration runs as a single instance and does not support vertical scaling or data slicing.
* Fetch Migration does not support customizing the list of indices included for migration.

> * Fetch migration only supports basic authentication.

Reviewer: Note - this line should be removed once https://github.com/opensearch-project/opensearch-migrations/pull/394 is merged

* The migration does not filter out indexes whose health is `red`.
* In the event that the migration fails or the process dies, the created indexes on the target cluster are not rolled back to the previous version.

## Running Fetch Migration

> Use either [Python](#python) or [Docker](#Docker) to run the Fetch Migration tool.

Reviewer 2: This tooling requires python3. We should call that out specifically.

> Use either [Python](#python) or [Docker](#Docker) to run the Fetch Migration tool.

Author: ```suggestion
Use either [python3](#python) or [Docker](#Docker) to run the Fetch Migration tool.
```

### Python

1. [Clone](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository) the `opensearch-migrations` repo using `git clone git@github.com:opensearch-project/opensearch-migrations.git`.

> 2. Install [Python](https://www.python.org/).

Reviewer 2: Python3

Author: Isn't `pip` apart of the python3 package?

> 2. Install [Python](https://www.python.org/).

Author: ```suggestion
2. Install [python3](https://www.python.org/downloads/).
```

3. Ensure that [pip](https://pip.pypa.io/en/stable/installation/#) by entering `pip --version`.
4. (Optional) Set up and activate a [virtual environment](https://packaging.python.org/en/latest/tutorials/installing-packages/#creating-and-using-virtual-environments).

> 5. Navigate to you cloned GitHub repo. Then, install the required Python dependencies by running the following command:

Reviewer: Nit - "your"

  ```shell
  python -m pip install -r python/requirements.txt
  ```

6. Inside the your cloned repository, run the orchestrator module using the following command:

  ```shell
  python python/fetch_orchestrator.py --help
  ```

When successful, the script returns confirmation that your indexes have been moved from the source cluster to the target cluster.

### Docker

To use the Fetch Migration tool with Docker, use the following steps:

> 1. From your cloned Git repository, build the `fetch-migration` Docker image using the f

Reviewer 2: The end of this looks incomplete

> 1. From your cloned Git repository, build the `fetch-migration` Docker image using the f

Author: ```suggestion
1. From your cloned Git repository, build the `fetch-migration` Docker image using the following command:
```
